src/main/java/com/google/api/codegen/discovery/viewmodel/SampleView.java

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.viewmodel;
+
+import com.google.api.codegen.SnippetSetRunner;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.auto.value.AutoValue;
+import java.util.List;
+
+@AutoValue
+public abstract class SampleView implements ViewModel {
+
+  @Override
+  public abstract String templateFileName();
+
+  @Override
+  public abstract String outputPath();
+
+  @Override
+  public String resourceRoot() {
+    // TODO(saicheems): Needed?
+    return SnippetSetRunner.SNIPPET_RESOURCE_ROOT;
```

Comment:
This is the package path that the snippet files (*.snip) are located relative to. For MVVM, it is always 'com/google/api/codegen'. We could remove the resourceRoot() method from ViewModel and all derived classes, and in CommonSnippetSetRunner, pass SnippetSetRunner.SNIPPET_RESOURCE_ROOT instead of input.resourceRoot().